File: LeetCode/88. Merge Sorted Array.py
# 数组归并，注意下把数组多余的位抹去就行
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        length = len(nums1)
        index1 = 0
        index2 = 0
        for _ in range(m):
            while(index2 < n and nums2[index2] < nums1[index1]):
                nums1.insert(index1, nums2[index2])
                index1 += 1
                index2 += 1
            index1 += 1
        while(index2 < n):
            nums1[index1] = nums2[index2] 
            index1 += 1
            index2 += 1
        while(len(nums1) > m + n):
            nums1.pop(-1)
                

File: LeetCode/88. Merge Sorted Array.cpp
// 挺简单的，把更大的从num2拷贝到nums1就行，前面的不用动
// 看了下最佳答案，有点复杂，没看懂什么意思，实际提交测试好像也没有更快，暂时不管
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        while(m > 0 && n > 0) {
            if (nums1[m - 1] < nums2[n - 1]) {
                nums1[n + m - 1] = nums2[n - 1];
                --n;
            }
            else {
                nums1[n + m - 1] = nums1[m - 1];
                --m;
            }
        }
        
        while(n > 0) {
            nums1[n - 1] = nums2[n - 1];
            --n;
        }
    }
};